GoogleForCreators/web-stories-wp
# Design System: Datalist a11y

## Context

Datalists were getting flagged for missing accessibility role on their containers. 

## Summary

We use a shared `<Datalist>` component for three areas: FontPicker, Author, Publisher Logo. Recently aXe started flagging the missing title for the container role modal here, so just fixing that to appease our e2e test aXe validation. 

## Relevant Technical Choices

- Add a title prop to the data list so that the role="modal" container has a title and aXe is happy. 
- The passed in aria-label was a little confusing, it actually only labels the collapsed select button, so renamed that to make it clearer what it is labelling.

## To-do



## User-facing changes

Now when you expand one of the above datalists, the div with the role 'modal' will have a title. 

And if you have aXe dev tools you can run it and make sure that this issue is resolved: 
<img width="1586" alt="Screen Shot 2022-03-21 at 3 26 39 PM" src="https://user-images.githubusercontent.com/10720454/159373494-8115b232-7709-47f8-af12-2197339a1b87.png">


## Testing Instructions

Should not see this error anymore: 
<img width="1003" alt="Screen Shot 2022-03-21 at 12 34 28 PM" src="https://user-images.githubusercontent.com/10720454/159373087-1b0df820-0b7b-4b66-bd5c-ef3eb6ded7ed.png">

See that datalists for author, publisher logos, font family all have dialogs with a 'title'. 

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10919
